gluonnlp/data/transforms.py

```diff
@@ -134,7 +136,45 @@ def __call__(self, sample):
                                           'mxnet.NDArray, received type=%s' % str(type(sample)))
 
 
-class NLTKMosesTokenizer(object):
+class _MosesTokenizer(object):
+    r"""Base class for moses tokenizer.
+    """
+    def __init__(self):
+        try:
+            from sacremoses import MosesTokenizer
+        except ImportError:
+            warnings.warn('MosesTokenizer has been deprecated in nltk 3.3.0. '
+                          'Please use sacremoses or older nltk version, e.g. 3.2.5. '
+                          'To install sacremoses, use pip install -U sacremoses')
+            try:
+                from nltk.tokenize.moses import MosesTokenizer
+            except ImportError:
+                raise ImportError('NLTK or relevant packages are not installed. '
+                                  'You must install NLTK <= 3.2.5 in order to use the '
+                                  'NLTKMosesTokenizer. You can refer to the official '
+                                  'installation guide in https://www.nltk.org/install.html .')
```

Comment:
This might not be right if invoked from SacreMosesTokenizer

Reply:
perhaps I will put the warning in the derived classes instead.

Reply:
Why do we need two classes? `NLTKMosesTokenizer` and `SacreMosesTokenizer`?

Reply:
Yes.